packages/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m

```diff
@@ -88,7 +88,18 @@ - (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNoti
   NSDictionary *remoteNotification = response.notification.request.content.userInfo;
   if (remoteNotification[@"gcm.message_id"]) {
     NSDictionary *notificationDict = [RNFBMessagingSerializer remoteMessageUserInfoToDict:remoteNotification];
-    [[RNFBRCTEventEmitter shared] sendEventWithName:@"messaging_notification_opened" body:notificationDict];
+
+    // create a merged dictionary that will hold the serialized userInfo data, as well as any actionIdentifier
+    NSMutableDictionary * mergedDict = [NSMutableDictionary dictionary];
+
+    // copy the values of notificationDict into mergedDict
+    [mergedDict addEntriesFromDictionary:notificationDict];
+
+    // if there is an actionIdentifier on this notification, also add that in.
+    [mergedDict setObject:response.actionIdentifier forKey:@"actionIdentifier"];
```

Comment:
I think this needs an addition to the RemoteMessage types so that actionIdentifier shows up here?

https://rnfirebase.io/reference/messaging/remotemessage#properties

https://github.com/invertase/react-native-firebase/blob/daaa72d82df6b9e5e2c1247c10792d4b12683541/packages/messaging/lib/index.d.ts#L69

Reply:
Ok I will sort this!

Reply:
gentle reminder: this needs a type update here